python/WhyLabs_Platform_with_log_reference.ipynb

```diff
@@ -0,0 +1,718 @@
+    "\n",
+    "Ensure you have a **model ID** (also called **dataset ID**) before you start!\n",
+    "\n",
+    "### Dataset Timestamp\n",
+    "* To avoid confusion, it's recommended that you use UTC\n",
+    "* If you don't set `dataset_timestamp` parameter, it'll default to `UTC` now\n",
+    "* WhyLabs supports real time visualization when the timestamp is **within the last 7 days**. Anything older than than will be picked up when we run our batch processing\n",
+    "* **If you log two profiles for the same day with different timestamps (12:00 vs 12:01), they are merged to the same batch**\n",
+    "\n",
+    "### Logging Different Batches of Data\n",
+    "* We'll give the profiles different **dates**\n",
+    "* Create a new logger for each date. Note that the logger needs to be `closed` to flush out the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3a006a2b-8403-477f-a1f5-a37ea33b160c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Enter your model ID from WhyLabs:\")\n",
+    "model_id = input()\n",
+    "reference_profile = None\n",
+    "for i, df in enumerate(pdfs):\n",
+    "    # walking backwards. Each dataset has to map to a date to show up as a different batch\n",
+    "    # in WhyLabs\n",
+    "    dt = datetime.datetime.now(tz=datetime.timezone.utc) - datetime.timedelta(days=i)\n",
+    "    \n",
+    "    # Create new logger for date\n",
+    "    with session.logger(tags={\"datasetId\": model_id}, dataset_timestamp=dt) as ylog:\n",
+    "        print(\"Log data frame for \", dt)\n",
+    "        ylog.log_dataframe(df)\n",
+    "        # we will keep a reference to the first profile for us as a baseline for monitoring\n",
+    "        if (i==0):\n",
+    "            reference_profile = ylog.profile"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "4b0ae3f5-395b-4a24-bde3-2779176ab9ee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Ensure everything is flushed\n",
+    "session.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3c3cfcf1",
+   "metadata": {},
+   "source": [
+    "We still have a reference to the first profile, for this demo we will use this dataframe's profile and upload it as a reference profile for monitoring on Whylabs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1fa34a96",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# You can rename the reference profile alias, this will show up when choosing a baseline on the monitoring settings page of Whylabs\n",
+    "reference_profile_alias = \"demo-reference-profile\"\n",
+    "reference_profile.to_summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7ad55e9e",
+   "metadata": {},
+   "source": [
+    "The reference profile can be uploading using a whylabs_client directly, first we need to reference the profile as a file on disk, so write it out."
```

Comment:
can be upload_ed_
directly. First, we need...

Reply:
yeah good catch, fixed.